flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1699,6 +1716,69 @@ public CatalogTableStatistics getPartitionStatistics(
         }
     }
 
+    @Override
+    public List<CatalogTableStatistics> bulkGetPartitionStatistics(
+            ObjectPath tablePath, List<CatalogPartitionSpec> partitionSpecs)
+            throws PartitionNotExistException, CatalogException {
+
+        return getTableAndPartitions(tablePath, partitionSpecs).f1.stream()
+                .map(p -> createCatalogTableStatistics(p.getParameters()))
+                .collect(Collectors.toList());
+    }
+
+    /** @return Partitions and names of partitions. */
+    private Tuple3<Table, List<Partition>, List<String>> getTableAndPartitions(
```

Comment:
We can just split this method into two methods, one is to get the HiveTable, and another is to get partitions. That could make the result type more simple and readable